hbase-server/src/main/resources/hbase-webapps/master/table.jsp

```diff
@@ -387,68 +388,93 @@ if (fqtn != null && master.isInitialized()) {
     }
   }
 %>
-<table class="table table-striped">
-  <tr>
-    <th>RegionName</th>
-    <th>Start Key</th>
-    <th>End Key</th>
-    <th>Replica ID</th>
-    <th>RegionState</th>
-    <th>ServerName</th>
-  </tr>
-<%
-  final boolean metaScanHasMore;
-  byte[] lastRow = null;
-  try (final MetaBrowser.Results results = metaBrowser.getResults()) {
-    for (final RegionReplicaInfo regionReplicaInfo : results) {
-      lastRow = Optional.ofNullable(regionReplicaInfo)
-        .map(RegionReplicaInfo::getRow)
-        .orElse(null);
-      if (regionReplicaInfo == null) {
-%>
-  <tr>
-    <td colspan="6">Null result</td>
-  </tr>
-<%
-      continue;
-    }
+<div style="overflow-x: auto">
+  <table class="table table-striped nowrap">
+    <tr>
+      <th title="Region name">RegionName</th>
+      <th title="The startKey of this region">Start Key</th>
+      <th title="The endKey of this region">End Key</th>
+      <th title="Region replica id">Replica ID</th>
+      <th title="State of the region while undergoing transitions">RegionState</th>
+      <th title="Server hosting this region replica, stored in info:server column">Server</th>
```

Comment:
It's a bit tedious, but how about we build all these labels from the values in `HConstants`? That way this page is captured by any future refactoring that happens down the line. For instance, the bit in this tool tip that says "info:server" would be replaced with `+ HConstants.CATALOG_FAMILY_STR + ":" + HConstants.SERVER_QUALIFIER_STR`.

Kind of fiddly. I'm not sure if it's worth it. Just a suggestion.

Reply:
I like the idea. I have changed that. To make this table header fit in one line, I have defined some column name variables, which can be reused.